File: rust/project/src/controllers/customer.rs
use actix_web::{get, post, put, delete, web, HttpResponse, Responder};
use sqlx::SqlitePool;
use crate::models::customer::Customer;

#[get("/customers")]
pub async fn get_customers(pool: web::Data<SqlitePool>) -> impl Responder {
    let result = sqlx::query_as::<_, Customer>(
        "SELECT * FROM customers"
    )
    .fetch_all(pool.get_ref())
    .await;

    match result {
        Ok(data) => HttpResponse::Ok().json(data),
        Err(_) => HttpResponse::InternalServerError().finish(),
    }
}

#[post("/customers")]
pub async fn create_customer(
    pool: web::Data<SqlitePool>,
    item: web::Json<Customer>,
) -> impl Responder {
    let res = sqlx::query(
        "INSERT INTO customers (full_name, email, phone) VALUES (?, ?, ?)"
    )
    .bind(&item.full_name)
    .bind(&item.email)
    .bind(&item.phone)
    .execute(pool.get_ref())
    .await;

    match res {
        Ok(_) => HttpResponse::Created().finish(),
        Err(_) => HttpResponse::InternalServerError().finish(),
    }
}

#[put("/customers/{id}")]
pub async fn update_customer(
    pool: web::Data<SqlitePool>,
    id: web::Path<i64>,
    item: web::Json<Customer>,
) -> impl Responder {
    let res = sqlx::query(
        "UPDATE customers SET full_name=?, email=?, phone=? WHERE id=?"
    )
    .bind(&item.full_name)
    .bind(&item.email)
    .bind(&item.phone)
    .bind(*id)
    .execute(pool.get_ref())
    .await;

    match res {
        Ok(_) => HttpResponse::Ok().finish(),
        Err(_) => HttpResponse::InternalServerError().finish(),
    }
}

#[delete("/customers/{id}")]
pub async fn delete_customer(
    pool: web::Data<SqlitePool>,
    id: web::Path<i64>,
) -> impl Responder {
    let res = sqlx::query("DELETE FROM customers WHERE id=?")
        .bind(*id)
        .execute(pool.get_ref())
        .await;

    match res {
        Ok(_) => HttpResponse::Ok().finish(),
        Err(_) => HttpResponse::InternalServerError().finish(),
    }
}

File: python/project/models/customer.py
from db import get_db

class Customer:
    @staticmethod
    def create_table():
        db = get_db()
        db.execute('''
            CREATE TABLE IF NOT EXISTS customers (
                id INTEGER PRIMARY KEY AUTOINCREMENT,
                full_name TEXT NOT NULL,
                email TEXT NOT NULL,
                phone TEXT NOT NULL,
                created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP
            )
        ''')
        db.commit()

    @staticmethod
    def get_all():
        db = get_db()
        cursor = db.execute('SELECT * FROM customers')
        return cursor.fetchall()

    @staticmethod
    def create(full_name, email, phone):
        db = get_db()
        db.execute(
            'INSERT INTO customers (full_name, email, phone) VALUES (?, ?, ?)',
            (full_name, email, phone)
        )
        db.commit()

    @staticmethod
    def update(id, full_name, email, phone):
        db = get_db()
        db.execute(
            'UPDATE customers SET full_name=?, email=?, phone=? WHERE id=?',
            (full_name, email, phone, id)
        )
        db.commit()

    @staticmethod
    def delete(id):
        db = get_db()
        db.execute('DELETE FROM customers WHERE id=?', (id,))
        db.commit()
